frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,675 @@
+          <SocialLink
+            href="https://github.com/inkofpixel"
+            color="rgba(24, 23, 23, 0.7)"
+            onHoverColor="rgba(24, 23, 23, 1)">
+            <GithubIcon color="white" />
+          </SocialLink>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+enum FormStatus {
+  Idle = "idle",
+  Submitting = "submitting",
+  Submitted = "submitted",
+}
+
+const blankForm = {
+  status: FormStatus.Idle,
+  values: { name: "", email: "", message: "" },
+  validationErrors: { name: null, email: null, message: null },
+  submitError: "",
+};
+
+interface UpdateFieldAction {
+  type: FormActionType.UpdateField;
+  name: string;
+  value: string;
+}
+
+interface SuccessAction {
+  type: FormActionType.Success;
+}
+
+interface FailAction {
+  type: FormActionType.Failed;
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+}
+
+type FormAction = UpdateFieldAction | SuccessAction | FailAction | SubmitAction;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: string;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: "",
+        },
+        submitError: "",
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      let error = false;
+      if (
+        state.values.name == "" ||
+        !state.values ||
+        state.values.name?.replace(/\s/g, "").length == 0
+      ) {
+        state.validationErrors.name = "Please insert your name";
+        error = true;
+      } else {
+        state.validationErrors.name = "";
+        if (
+          state.values.email == "" ||
+          !state.values.email ||
+          state.values.email?.replace(/\s/g, "").length == 0
+        ) {
+          state.validationErrors.email = "Please insert your email";
+          error = true;
+        } else {
+          if (!regex.test(state.values.email)) {
+            state.validationErrors.email = "Please insert a valid email";
+            error = true;
+          } else {
+            state.validationErrors.email = "";
+            if (state.values.message == "" || !state.values.message) {
+              state.validationErrors.message = "Please insert a message";
+              error = true;
+            } else {
+              state.validationErrors.message = "";
+            }
+          }
+        }
+      }
```

Comment:
condition statements that validates fields, should not be nested, otherwise you are only reporting one error at a time, and this is bad UX (when I submit, I want to know about every errors I've made, so that I can try to correct them in one pass)